tests/include/detray/test/device/sycl/CMakeLists.txt

```diff
@@ -0,0 +1,28 @@
+# Detray library, part of the ACTS project (R&D line)
+#
+# (c) 2024 CERN for the benefit of the ACTS project
+#
+# Mozilla Public License Version 2.0
+
+# Set the SYCL build flags.
+include(detray-compiler-options-sycl)
+
+# Enable SYCL as a language.
+enable_language(SYCL)
+
+include_directories( SYSTEM $<TARGET_PROPERTY:covfie::core,INTERFACE_INCLUDE_DIRECTORIES> )
+
+# Set up a test library, which the "new style" benchmarks and tests could use.
+file( GLOB _detray_test_headers
+   RELATIVE "${CMAKE_CURRENT_SOURCE_DIR}"
+   "../propagator_test.hpp")
+
+add_library( detray_test_sycl INTERFACE "${_detray_test_headers}" )
```

Comment:
I'm really not in love with this setup. :frowning:

You're picking up a header, that is shared with the CUDA code. Yet, you put 2 different libraries around that one header. :thinking:

And the linked library list doesn't even make much sense. :frowning: You link `detray_test_sycl` against `vecmem::sycl`. Yet, there is absolutely nothing SYCL specific in `propagator_test.hpp`. It does however directly depend on `vecmem::core`, `covfie::core` and `GTest::gtest`. (On top of all of its Detray dependencies, which I didn't double-check.)

You should create a `detray_test_device` `IMPORTED` library for the shared header(s) instead. And for now just get rid of `detray_test_sycl`, until it will actually have some SYCL code. :thinking:

Reply:
I added it as an interface library now, not sure where to set the ```IMPORTED``` thing. It seems to be for external libraries?

Reply:
That was my typo. I meant `INTERFACE`, not `IMPORTED`. :wink:

Reply:
Perfect, then thats done :)